lighthouse-core/test/computed/resource-summary-test.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const ComputedResourceSummary = require('../../computed/resource-summary.js');
+const assert = require('assert');
+
+/* eslint-env jest */
+
+function mockTracingData(recordInfo) {
+  const networkRecords = recordInfo.map((info, index) => ({
+    requestId: index.toString(),
+    resourceType: info.resourceType,
+    frameId: 1,
+    finished: true,
+    priority: 'HIGH',
+    initiatorRequest: null,
+    statusCode: 200,
+    transferSize: info.transferSize,
+    url: info.url,
+  }));
+
+  return networkRecords;
+}
+
+describe('Resource summary computed', () => {
+  let networkRecords;
+  let mainResource;
+  beforeEach(() => {
+    const networkRecordInfo = [
+      {url: 'http://example.com/file.html', resourceType: 'Document', transferSize: 30},
+      {url: 'http://example.com/app.js', resourceType: 'Script', transferSize: 10},
```

Comment:
not necessary but might be cool to make this `http://cdn.example.com/app.js` so reader knows it's not just origin comparison that's a `third-party`

Reply:
It's just origin comparison 😂. I know we talked about incorporating your 3rd party analysis here, but since that's probably not a trivial addition, I figured this would do for now?

Reply:
i guess patrick is suggesting that script.js move to cdn.example.com so the case of subdomains is handled in the test (and the total is captured in 3P). i like that.

Reply:
I updated the tests covering 3P calculations. Let me know if this matches what you had in mind.